src/replication.c

```diff
@@ -3053,6 +3061,8 @@ void replicationSetMaster(char *ip, int port) {
     server.repl_state = REPL_STATE_CONNECT;
     serverLog(LL_NOTICE,"Connecting to MASTER %s:%d",
         server.masterhost, server.masterport);
+    /* Master link changed, reset the attempts number. */
+    server.repl_master_sync_attempts = 0;
```

Comment:
i think this reset is also needed in `replicationHandleMasterDisconnection()` if we were previously connected.
i.e.
```c
    if (server.repl_state == REPL_STATE_CONNECTED)
        server.repl_current_attempts = 0;
```

Reply:
```
[replicaof ip port] -> [connect to master] -> [success] -> [maybe disconnect and reconnect] -> [replicaof no one or new_ip new_port]
```

I hope to gather statistics for the entire cycle from when a replica starts establishing a master link until it becomes a master or switches to another master, rather than just the process from one disconnection and reconnection to success.

Reply:
i don't like it.
it means that this counter can include incidents from the distant past, rather than the current network issue.

ie. imagine that you have some network issue, and you resolve it... then months later you have another network issue.
with the way you suggest, the user needs to monitor this counter over time and see if it's growing, or it's static.
with the way i suggest, a single look at the counter means that there's a network issue at present time, i.e. any number greater than one indicates an issue, and you don't need to watch the counter growing to judge.